### DISCUSSION TITLE
"Cockpit is not compatible with the software on the system" when adding welcome message through .bashrc file

### DISCUSSION THREAD

**Discussion by ankeshanand1** [cockpit-project/cockpit#18363] [Q&A] [answered]
### Explain what happens

1. I have a multi-server setup with cockpit. My main server which is running cockpit is on Ubuntu 20.04 while some of them are on CentOS7.9.
2. I recently configured a welcome message through `.bashrc` file where I added the following lines in the file:
```
if [ -f ~/.Banner.sh ]; then
    source ~/.Banner.sh
fi
```
3. The `.Banner.sh` file contains some welcome message which is displayed through `echo`. This has been recently implemented on my main servers and some other servers.
4. Though the terminal on main server(through cockpit) is opening perfectly, the other servers are having connection problems with the main cockpit server and it's showing "Cockpit is not compatible with the software on the system"
5. The `.Banner.sh` file echoes a welcome message which shows Operating system name and other hardware configs, nothing related to cockpit which may indicate some error. 

### Version of Cockpit

215-1

### Where is the problem in Cockpit?

Unknown or not applicable

### Server operating system

Ubuntu

### Server operating system version

Ubuntu 20.04.5 LTS

### What browsers are you using?

Chrome

### System log

```shell
Feb 17 02:51:44 mainserver.hostname cockpit-bridge[52464]: /usr/lib/cockpit/cockpit-ssh: incorrect protocol: received invalid length prefix
Feb 17 02:51:48 mainserver.hostname sshd[58110]: error: kex_exchange_identification: read: Connection reset by peer
```

**Comment by martinpitt** [comment, marked as answer]
Such banners tend to break automation a lot if they happen for non-interactive logins. If you do `ssh yourmachine whoami`, do you see the banner? If so, I recommend to wrap the sourcing into the usual "is interactive?" check:
```sh
if [ -n "$PS1" ] && [ -f ~/.Banner.sh ]; then
    source ~/.Banner.sh
fi
```

Other than that, we actually circumvented such noise in commit 87151a68e59f9f in version 196. This commit is also in RHEL/CentOS 7.9. Do you see a pattern that it works when connecting to other Ubuntu machines, but it fails with the CentOS ones?

In other words, I cannot reproduce this -- we have integration tests that do exactly this "stdout noise" and it works; so either this banner.sh does something more special, or there's something else which triggers this error. That said, both of these versions are really old. Ubuntu 20.04 has a newer backport, but newer versions aren't supported on RHEL/CentOS 7.

**Reply by ankeshanand1** [reply]
Thank you very much!
Adding `[ -n "$PS1" ] &&` did the trick
